sdk/identity/azure-identity/azure/identity/_internal/managed_identity_client.py

```diff
@@ -84,8 +84,8 @@ def _process_response(self, response, request_time):
 
         return token
 
-    def get_cached_token(self, *scopes):
-        # type: (*str) -> Optional[AccessToken]
+    def get_cached_token(self, *scopes, **kwargs):   # pylint:disable=unused-argument
```

Comment:
I think this is more complicated than the change itself:

- Service Principal credentials support `data`, but 
- Managed Identity credentials don't. (They may support `data` in the future.)

Making them share the same `GetTokenMixin` adds unused `kwargs` to `ManagedIdentityClientBase.get_cached_token`.